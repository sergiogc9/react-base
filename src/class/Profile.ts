export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
