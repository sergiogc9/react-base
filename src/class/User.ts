export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
