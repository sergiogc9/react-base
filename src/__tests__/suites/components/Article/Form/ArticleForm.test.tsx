import { mount } from "enzyme";
import React from "react";
import { MemoryRouter } from 'react-router-dom';

import ArticleForm from "@src/components/Article/Form/ArticleForm";
import GA from '@src/lib/googleAnalytics';
import { TenantObject, Settings } from "@src/class/Tenant";

const tenantSettings: Settings = {
	categorization_mode: 'flc',
	currency: 'EUR',
	display_influencers: false,
	document_tenant_indices_search: true,
	facebook_url: 'fake url',
	valuation_metric: 'miv'
};

const tenant: TenantObject = {
	id: 'rd-girona-test',
	guid: '00034972-0000-0000-0000-000000000000',
	name: 'rd.girona.test',
	tier_properties: {
		name: 'diamond',
		limit: 50000,
		results: {
			online: true,
			social: true
		}
	},
	print_only: false,
	facebook_linked_ids: [
		'10156337492683158',
		'10160315799795215',
		'206028773584551',
		'611466659195813'
	],
	settings: tenantSettings
};

describe("Search Form", () => {
	let wrapper: any;

	let onSetSortMock = jest.fn();
	let onSetLimitMock = jest.fn();
	let onSetQueryMock = jest.fn();
	let onSubmitMock = jest.fn();
	let onToggleDrawer = jest.fn();
	let onSetShowInsertDocument = jest.fn();

	const getArticleFormComponent = () =>
		(
			<ArticleForm
				query=""
				period="last_week"
				sort="publication_date:desc"
				limit={20}
				start={0}
				total={200}
				onSubmit={onSubmitMock}
				onToggleDrawer={onToggleDrawer}
				onSetQuery={onSetQueryMock}
				onSetLimit={onSetLimitMock}
				onSetSort={onSetSortMock}
				onSetShowInsertDocument={onSetShowInsertDocument}
				loadingDocuments={false}
				tenant={tenant}
			/>
		);

	const getComponentMountedWithMemoryRouter = () => mount(
		<MemoryRouter>
			{getArticleFormComponent()}
		</MemoryRouter>
	);

	beforeEach(() => {

		onSetSortMock = jest.fn();
		onSetLimitMock = jest.fn();
		onSetQueryMock = jest.fn();
		onSubmitMock = jest.fn();
		onToggleDrawer = jest.fn();
		onSetShowInsertDocument = jest.fn();

		wrapper = mount(getArticleFormComponent());
	});

	it("Test component snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test submit", () => {
		GA.trackEvent = jest.fn();
		expect(GA.trackEvent).toHaveBeenCalledTimes(0);
		expect(onSubmitMock).toHaveBeenCalledTimes(0);
		wrapper.find("button.form-search-icon").at(0).simulate('click');
		expect(onSubmitMock).toHaveBeenCalledTimes(1);
		expect(GA.trackEvent).toHaveBeenCalledTimes(1);
	});

	it("Test sort by echo", () => {
		GA.trackEvent = jest.fn();
		expect(GA.trackEvent).toHaveBeenCalledTimes(0);
		expect(onSetSortMock).toHaveBeenCalledTimes(0);
		wrapper.find("#sortByInputSelect").at(0).props().onChange("echo:desc");
		expect(onSetSortMock).toHaveBeenCalledTimes(1);
	});

	it("Test set query", () => {
		expect(onSetQueryMock).toHaveBeenCalledTimes(0);
		wrapper.find("#searchFormInputText").at(0).props().onChange("query");
		expect(onSetQueryMock).toHaveBeenCalledTimes(1);
	});

	it("Test submit from query input", () => {
		expect(onSubmitMock).toHaveBeenCalledTimes(0);
		wrapper.find("input#searchFormInputText").simulate('keydown', { keyCode: 13 });
		expect(onSubmitMock).toHaveBeenCalledTimes(1);
	});

	it("Test submit from query input without submit", () => {
		expect(onSubmitMock).toHaveBeenCalledTimes(0);
		wrapper.find("input#searchFormInputText").simulate('keydown', { keyCode: 14 });
		expect(onSubmitMock).toHaveBeenCalledTimes(0);
	});

	it("Test set limit", () => {
		wrapper = getComponentMountedWithMemoryRouter();

		GA.trackEvent = jest.fn();

		expect(GA.trackEvent).toHaveBeenCalledTimes(0);
		expect(onSetLimitMock).toHaveBeenCalledTimes(0);
		wrapper.find("button#formOptions-toggle").at(0).simulate('click');
		wrapper.find("ul#formOptions-list li").at(3).find("div.md-fake-btn").simulate('click');
		expect(onSetLimitMock).toHaveBeenCalledTimes(1);

		wrapper.find("button#formOptions-toggle").at(0).simulate('click');
		wrapper.find("ul#formOptions-list li").at(4).find("div.md-fake-btn").simulate('click');
		expect(onSetLimitMock).toHaveBeenCalledTimes(2);

		wrapper.find("button#formOptions-toggle").at(0).simulate('click');
		wrapper.find("ul#formOptions-list li").at(5).find("div.md-fake-btn").simulate('click');
		expect(onSetLimitMock).toHaveBeenCalledTimes(3);

		wrapper.find("button#formOptions-toggle").at(0).simulate('click');
		wrapper.find("ul#formOptions-list li").at(6).find("div.md-fake-btn").simulate('click');
		expect(onSetLimitMock).toHaveBeenCalledTimes(4);

		expect(GA.trackEvent).toHaveBeenCalledTimes(4);
	});

	it("Test on set show insert document click", () => {
		wrapper = getComponentMountedWithMemoryRouter();

		expect(onSetShowInsertDocument).toHaveBeenCalledTimes(0);
		wrapper.find("button#formOptions-toggle").at(0).simulate('click');
		wrapper.find("ul#formOptions-list li").at(1).find("div.md-fake-btn").simulate('click');
		expect(onSetShowInsertDocument).toHaveBeenCalledTimes(1);
	});

	it("Test click drawer", () => {
		expect(onToggleDrawer).toHaveBeenCalledTimes(0);
		wrapper.find("button#searchFormToggleDrawerBtn").simulate('click');
		expect(onToggleDrawer).toHaveBeenCalledTimes(1);
	});

	it("Test select sort by from button", () => {
		expect(onSetSortMock).toHaveBeenCalledTimes(0);
		wrapper.find("button#formSortByButton-toggle").at(0).simulate('click');
		wrapper.find("ul#formSortByButton-list li").at(0).find("div.md-fake-btn").simulate('click');
		expect(onSetSortMock).toHaveBeenCalledTimes(1);

		wrapper.find("button#formSortByButton-toggle").at(0).simulate('click');
		wrapper.find("ul#formSortByButton-list li").at(1).find("div.md-fake-btn").simulate('click');
		expect(onSetSortMock).toHaveBeenCalledTimes(2);

		wrapper.find("button#formSortByButton-toggle").at(0).simulate('click');
		wrapper.find("ul#formSortByButton-list li").at(2).find("div.md-fake-btn").simulate('click');
		expect(onSetSortMock).toHaveBeenCalledTimes(3);

		wrapper.find("button#formSortByButton-toggle").at(0).simulate('click');
		wrapper.find("ul#formSortByButton-list li").at(3).find("div.md-fake-btn").simulate('click');
		expect(onSetSortMock).toHaveBeenCalledTimes(4);
	});

	it("Test manage feeds tracker", () => {
		wrapper = getComponentMountedWithMemoryRouter();

		GA.trackEvent = jest.fn();
		expect(GA.trackEvent).toHaveBeenCalledTimes(0);
		wrapper.find("button#formOptions-toggle").at(0).simulate('click');
		wrapper.find("ul#formOptions-list li").at(0).find("a.md-fake-btn").simulate('click');
		expect(GA.trackEvent).toHaveBeenCalledTimes(1);
	});

	it("Test audience option sort", () => {
		expect(wrapper.find("#sortByInputSelect").at(0).props().menuItems[0].value).toEqual('reach:desc');
	});
});
