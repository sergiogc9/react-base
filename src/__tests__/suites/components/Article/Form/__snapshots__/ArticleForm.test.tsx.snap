// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form Test component snapshot 1`] = `"<div id=\\"articleForm\\"><button id=\\"searchFormToggleDrawerBtn\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">filter_list</i></button><div class=\\"search-form-sort-by\\"><div id=\\"sortByInputSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field sort-by-select\\"><div class=\\"md-select-field__toggle\\"><label for=\\"sortByInputSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">results.sort.value</label><div id=\\"sortByInputSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">results.sort.publication_date</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"sortByInputSelect\\" value=\\"publication_date:desc\\"></div></div></div><div id=\\"formSortByButton\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu-container--menu-below form-sort-by-button\\"><button id=\\"formSortByButton-toggle\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon icon-documents_sort md-text--inherit\\"></i></button></div><button type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-btn--tooltip md-inline-block form-sort-by-help\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-tooltip-container\\"><span aria-hidden=\\"true\\"></span></div><i class=\\"md-icon material-icons md-text--inherit\\">help_outline</i></button></div><div id=\\"searchFormInput\\"><div class=\\"md-text-field-container md-full-width md-text-field-container--input\\"><label for=\\"searchFormInputText\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">form.search.documents</label><input id=\\"searchFormInputText\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-full-width md-text\\" value=\\"\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div><button type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block form-search-icon\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">search_outline</i></button></div><div class=\\"form-display-items\\"><div class=\\"form-results-display-label\\">results.search.documents.pagination_label</div><div class=\\"form-results-display-content\\">results.search.documents.pagination_info</div></div><div id=\\"formOptions\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container form-options-select\\"><button id=\\"formOptions-toggle\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">settings arrow_drop_down</i><div class=\\"md-icon-separator md-icon-text\\"><span class=\\"md-icon-text\\"></span></div></div></button></div></div>"`;
